From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 16 Nov 2017 23:43:29 +0100
Subject: [PATCH] {PRINTF} bpf: add __printf attribute to bpf_insn_print_cb
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc -Wsuggest-attribute=format complains:

    kernel/bpf/verifier.c: In function ‘do_check’:
    kernel/bpf/verifier.c:3819:19: error: argument 1 of ‘print_bpf_insn’
    might be a candidate for a format attribute
    [-Werror=suggest-attribute=format]
        print_bpf_insn(verbose, env, insn,
                       ^~~~~~~
---
 kernel/bpf/disasm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/bpf/disasm.h b/kernel/bpf/disasm.h
index 8de977e420b6..ad5425324e19 100644
--- a/kernel/bpf/disasm.h
+++ b/kernel/bpf/disasm.h
@@ -24,7 +24,7 @@ extern const char *const bpf_class_string[8];
 const char *func_id_name(int id);
 
 struct bpf_verifier_env;
-typedef void (*bpf_insn_print_cb)(struct bpf_verifier_env *env,
+typedef void (__printf(2, 3) *bpf_insn_print_cb)(struct bpf_verifier_env *env,
 				  const char *, ...);
 void print_bpf_insn(bpf_insn_print_cb verbose, struct bpf_verifier_env *env,
 		    const struct bpf_insn *insn, bool allow_ptr_leaks);
-- 
